src/RpcServer.cpp

```diff
@@ -84,7 +84,12 @@ bool RpcServer::PollRpc(bool blocking, double time_out, RpcCallInfo* call_info)
   }
 
   auto& item = m_poll_queue.front();
-  unsigned int call_uid = (item.conn_id << 16) | item.msg->seq_num_uid();
+  unsigned int call_uid;
+  // do not include conn id if the result came from the server
+  if (item.conn_id != 0xffff)
+    call_uid = (item.conn_id << 16) | item.msg->seq_num_uid();
+  else
+    call_uid = item.msg->seq_num_uid();  
```

Comment:
This changes both call_info->call_uid and the response map call_uid.  Do both of these need to be changed?  Also, this makes conn_id=0 and conn-id=0xffff equivalent, does this cause any issues anywhere else?